src/testdrive/parser.rs

```diff
@@ -120,16 +136,37 @@ fn parse_sql(line_reader: &mut LineReader) -> Result<SqlCommand, InputError> {
                 expected_rows.push(split_line(pos3, &line3)?);
             }
         }
-        (Some((pos2, line2)), None) => expected_rows.push(split_line(pos2, &line2)?),
+        (Some((pos2, line2)), None) => match HASH_REGEX.captures(&line2) {
+            Some(captures) => match captures.get(1).unwrap().as_str().parse::<usize>() {
+                Ok(num_values) => {
+                    return Ok(SqlCommand {
+                        query,
+                        expected_result: SqlExpectedResult::Hashed {
+                            num_values,
+                            md5: captures.get(2).unwrap().as_str().to_owned(),
+                        },
+                    })
+                }
+                Err(_) => {
+                    return Err(InputError {
+                        pos: pos2,
+                        msg: "Invalid value for number of expected rows".into(),
+                    });
+                }
+            },
+            None => expected_rows.push(split_line(pos2, &line2)?),
+        },
```

Comment:
`regex::Captures` implements `Index` which makes it quite a bit nicer to work with when you want to access str matches that you know to exist:

```diff
diff --git a/src/testdrive/parser.rs b/src/testdrive/parser.rs
index 8f7a4fcf..6a0449a3 100644
--- a/src/testdrive/parser.rs
+++ b/src/testdrive/parser.rs
@@ -137,13 +137,13 @@ fn parse_sql(line_reader: &mut LineReader) -> Result<SqlCommand, InputError> {
             }
         }
         (Some((pos2, line2)), None) => match HASH_REGEX.captures(&line2) {
-            Some(captures) => match captures.get(1).unwrap().as_str().parse::<usize>() {
+            Some(captures) => match captures[1].parse::<usize>() {
                 Ok(num_values) => {
                     return Ok(SqlCommand {
                         query,
                         expected_result: SqlExpectedResult::Hashed {
                             num_values,
-                            md5: captures.get(2).unwrap().as_str().to_owned(),
+                            md5: captures[2].to_owned(),
                         },
                     })
                 }
```